### ISSUE TITLE
No reportIncompatibleMethodOverride on base method with *args and **kwargs

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#11092]
### Discussed in https://github.com/microsoft/pyright/discussions/11091

<div type='discussions-op-text'>

<sup>Originally posted by **rchl** November  4, 2025</sup>
Not sure if it's a bug, limitation or on purpose so asking here first.

With a code like:

```py
from typing import Any, override


class Base:
    def run(self, foo: str, *args: Any, **kwargs: Any) -> None:
        pass


class Impl1(Base):
    def run(self, foo: int) -> None:
        pass


class Impl2(Base):
    def run(self) -> None:
        pass


class Impl3(Base):
    def run(self, foo: str) -> None:
        pass
```

pyright doesn't report `reportIncompatibleMethodOverride` in cases `Impl1` and `Impl2`. `mypy` does report:

```
_test_types.py:11: error: Signature of "run" incompatible with supertype "Base"  [override]
_test_types.py:11: note:      Superclass:
_test_types.py:11: note:          def run(self, foo: str, *args: Any, **kwargs: Any) -> None
_test_types.py:11: note:      Subclass:
_test_types.py:11: note:          def run(self, foo: int) -> None
_test_types.py:16: error: Signature of "run" incompatible with supertype "Base"  [override]
_test_types.py:16: note:      Superclass:
_test_types.py:16: note:          def run(self, foo: str, *args: Any, **kwargs: Any) -> None
_test_types.py:16: note:      Subclass:
_test_types.py:16: note:          def run(self) -> None
```
</div>
